embark/embark/consumers.py

```diff
@@ -0,0 +1,173 @@
+import json
+import logging
+
+import os
+import channels
+import rx
+from rx import Observable
+import rx.operators as ops
+import re
+from inotify_simple import flags
+from . import inotify_wrap
+import difflib
+from channels.generic.websocket import WebsocketConsumer
+from pathlib import Path
+
+
+# consumer class for synchronous/asynchronous websocket communication
+# TODO: Implement frontend Websocket handling and data extraction (Example in progress.html -> remove later)
+class WSConsumer(WebsocketConsumer):
+
+    # constructor
+    def __init__(self):
+        super().__init__()
+        # self.room_group_name = 'status_updates_group'
+        # global module count and status_msg directory
+        self.module_count = 0
+        # TODO: extend dictionary for more information
+        self.status_msg = {
+            "percentage": 0.0,
+            "module": "",
+            "phase": "",
+        }
+
+    # this method is executed when the connection to the frontend is established
+    def connect(self):
+        # accept socket connection
+        self.accept()
+
+        # if file does not exist create it otherwise delete its content
+        open('/app/emba/log_1/emba_new.log', 'w+')
```

Comment:
Are we sure the path will always be this?

Reply:
no its a placeholderfor testing purposes , but will be adjusted before the midterm release.